chore: Update Node.js compatibility doc (#17321)
newrelic/docs-website: src/content/docs/apm/agents/nodejs-agent/getting-started/compatibility-requirements-nodejs-agent.mdx

</Collapser>
</CollapserGroup>

{/* begin: compat-table */}

> ## Instrumented Modules 

Reviewer: ```suggestion
## Instrumented modules [#instrumented-modules] 
```

Author: I purposefully removed the `[#instrumented-modules]` bit because it is not valid markdown. If we were going to include it, we'd need to include it in our tool that generates all of the text between the markers. That same tool generates the document in our agent repository (linked in the opening statement of this issue). I'd prefer that we do as little customization of the generated output based on the target repository (docs vs agent) as possible.

Reviewer: Okay sure, we can remove the anchor name in this instance.

> After installation, the agent automatically instruments with our catalog of supported Node.js libraries and frameworks. This gives you immediate access to granular information specific to your web apps and servers.  For unsupported frameworks or libraries, you'll need to instrument the agent yourself using the [Node.js agent API](https://docs.newrelic.com/docs/apm/agents/nodejs-agent/api-guides/nodejs-agent-api/).

Reviewer: ```suggestion
After installation, our agent automatically instruments with a catalog of supported Node.js libraries and frameworks. This gives you immediate access to granular information specific to your web apps and servers. For unsupported frameworks or libraries, you'll need to instrument the agent yourself using the [Node.js agent API](/docs/apm/agents/nodejs-agent/api-guides/nodejs-agent-api/).
```

Author: As stated in the other review comment, all of this text is also replicated in the doc included in our agent repository. So is it a definite requirement that URL be relative instead of absolute?

Reviewer: Sure, we can include the absolute link in this situation, similar to exceptions we make for automatically generated release notes.

**Note**: The latest supported version may not reflect the most recent supported version.

> | Package Name | Minimum Supported Version | Latest Supported Version | Introduced In* |

Reviewer: ```suggestion
| Package name | Minimum supported version | Latest supported version | Introduced in* |
```

| apollo-server-koa | 2.14.0 | 3.13.0 | @newrelic/apollo-server-plugin@1.0.0 |
| apollo-server-lambda | 2.14.0 | 3.13.0 | @newrelic/apollo-server-plugin@1.0.0 |
| aws-sdk | 2.2.48 | 2.1620.0 | 6.2.0 |
| bluebird | 2.0.0 | 3.7.2 | 1.27.0 |
| bunyan | 1.8.12 | 1.8.15 | 9.3.0 |
| cassandra-driver | 3.4.0 | 4.7.2 | 1.7.1 |
| connect | 2.0.0 | 3.7.0 | 2.6.0 |
| director | 1.2.0 | 1.2.8 | 2.0.0 |
| express | 4.6.0 | 4.19.2 | 2.6.0 |
| fastify | 2.0.0 | 4.27.0 | 8.5.0 |
| generic-pool | 2.4.0 | 3.9.0 | 0.9.0 |
| ioredis | 3.0.0 | 5.4.1 | 1.26.2 |
| koa | 2.0.0 | 2.15.3 | 3.2.0 |
| koa-route | 2.0.0 | 4.0.1 | 3.2.0 |
| koa-router | 2.0.0 | 12.0.1 | 3.2.0 |
| mongodb | 2.1.0 | 6.6.1 | 1.32.0 |
| mysql | 2.2.0 | 2.18.1 | 1.32.0 |
| mysql2 | 1.3.1 | 3.9.7 | 1.32.0 |
| next | 13.0.0 | 14.2.3 | @newrelic/next@0.7.0 |
| openai | 4.0.0 | 4.47.1 | 11.13.0 |
| pg | 8.2.0 | 8.11.5 | 9.0.0 |
| pg-native | 2.0.0 | 3.0.1 | 9.0.0 |
| pino | 7.0.0 | 9.1.0 | 8.11.0 |
| redis | 2.0.0 | 4.6.13 | 1.31.0 |
| restify | 5.0.0 | 11.1.0 | 2.6.0 |
| superagent | 2.0.0 | 9.0.2 | 4.9.0 |
| undici | 4.7.0 | 6.16.1 | 11.1.0 |
| winston | 3.0.0 | 3.13.0 | 8.11.0 |

> *When package is not specified, support is within the `newrelic` package.

Reviewer: @jsumners-nr This sentence is a bit vague. Can you clarify what you mean?

Author: This is the footnote indicated on the "Introduced In" column heading. Notice that the column primarily lists version numbers, but has fields that exceptionally list a package.

        </td>

        <td>`next`</td>

        <td>
          12.0.9 - 13.3.0
        </td>

        <td>
          10.1.0
        </td>

        <td>
          v12.2.0 or higher required for middleware instrumentation
        </td>
      </tr>

    </tbody>
  </table>

If you are using a supported framework with default routers, the Node.js agent can read these frameworks' route names as is. However, if you want more specific names than are provided by your framework, you may want to use one or more of the tools New Relic provides with the [Node.js transaction naming API](/docs/nodejs/nodejs-transaction-naming-api).

  <Callout title="EOL NOTICE">
    We've discontinued support for several capabilities in November 2021. This includes the Oracle Driver Package and Hapi versions prior to Hapi 19.2 for our Node.js agent. For more details, including how you can prepare for this transition, see our [Support Forum post](https://discuss.newrelic.com/t/important-upcoming-changes-to-capabilities-and-support-across-node-agents-suggested-pagerduty-responders-incident-workflows-and-kubernetes-instrumentation/164481?u=dholloran).
  </Callout>
</Collapser>
</CollapserGroup>

> ## Connect the agent to other New Relic features [#digital-intelligence-platform]

Reviewer 2: this section will be needed. Some may be cleaned up but other links are unrelated to libraries we support
